internal/cli/certificate.go

```diff
@@ -0,0 +1,367 @@
+	fccrypto "github.com/flightctl/flightctl/internal/crypto"
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"golang.org/x/term"
+	"sigs.k8s.io/yaml"
+)
+
+const secondsInDay = 86400
+
+type CertificateOptions struct {
+	GlobalOptions
+	Name            string
+	SignerName      string
+	Expiration      string
+	OutputFormat    string
+	PrivKeyPassword string
+}
+
+func DefaultCertificateOptions() *CertificateOptions {
+	return &CertificateOptions{
+		GlobalOptions: DefaultGlobalOptions(),
+		SignerName:    "ca",
+		OutputFormat:  "components",
+	}
+}
+
+func NewCmdCertificate() *cobra.Command {
+	o := DefaultCertificateOptions()
+	cmd := &cobra.Command{
+		Use: "certificate [subcommand]",
+		// more subcommands may be added later
+		Short:     "request a new enrollment certificate for a device with 'certificate request'",
+		Args:      cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
+		ValidArgs: []string{"request"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := o.Validate(args); err != nil {
+				return err
+			}
+			return o.Run(cmd.Context(), args)
+		},
+		SilenceUsage: true,
+	}
+	o.Bind(cmd.Flags())
+	return cmd
+}
+
+func (o *CertificateOptions) Bind(fs *pflag.FlagSet) {
+	o.GlobalOptions.Bind(fs)
+	fs.StringVarP(&o.Name, "name", "n", "mycsr", "specify a name for the certificate signing request")
+	fs.StringVarP(&o.SignerName, "signer", "s", "ca", "specify certificate signer - options currently supported: 'ca'")
+	fs.StringVarP(&o.Expiration, "expiration", "e", "7d", "specify desired certificate expiration in days, example: 7d")
+	fs.StringVarP(&o.OutputFormat, "output", "o", "components", "specify desired output format: either 'agent-config' to output values in one config file, or 'components' to output cert, key, and ca values in separate files")
+	fs.StringVarP(&o.PrivKeyPassword, "password", "w", "", "specify a password for newly generated private key, which will be encrypted on local disk")
+}
+
+func (o *CertificateOptions) Complete(cmd *cobra.Command, args []string) error {
+	return o.GlobalOptions.Complete(cmd, args)
+}
+
+func (o *CertificateOptions) Validate(args []string) error {
+	if err := o.GlobalOptions.Validate(args); err != nil {
+		return err
+	}
+
+	re := `^\d+d$`
+	matched, err := regexp.MatchString(re, o.Expiration)
+	if err != nil || !matched {
+		return fmt.Errorf("invalid expiration specified: %s\nexpiration must be in the form of days, example: 365d", o.Expiration)
+	}
+
+	return nil
+}
+
+func (o *CertificateOptions) Run(ctx context.Context, args []string) error {
+	fmt.Printf("the expiration is %s\n", o.Expiration)
+
+	name := createUniqueName(o.Name)
+
+	fmt.Println("this operation will generate a new key pair")
+	fmt.Println("creating new key pair...")
+	_, priv, err := fccrypto.NewKeyPair()
+	if err != nil {
+		return fmt.Errorf("error creating new key pair: %w", err)
```

Comment:
Cobra itself will prefix `Error:`, so you can leave out the `error ` from the messages.